graphql-cli/src/test/java/io/ballerina/graphql/generator/ballerina/ConfigTypesGeneratorTest.java

```diff
@@ -0,0 +1,175 @@
+                            "graphql.config.yaml")).toString(),
+                    this.tmpDir);
+
+            Extension extensions = projects.get(0).getExtensions();
+
+            AuthConfig authConfig = new AuthConfig();
+            AuthConfigGenerator.getInstance().populateAuthConfigTypes(extensions, authConfig);
+            AuthConfigGenerator.getInstance().populateApiHeaders(extensions, authConfig);
+
+            ConfigGenerator.getInstance().generateConfigRecord(authConfig);
+
+            String generatedConfigTypesContent = ConfigGenerator.getInstance().generateSrc(authConfig)
+                    .trim().replaceAll("\\s+", "")
+                    .replaceAll(System.lineSeparator(), "");
+
+            Path expectedConfigTypesFile =
+                    resourceDir.resolve(Paths.get("expectedGenCode", "client",
+                            "config_types.bal"));
+            String expectedUtilsContent = readContent(expectedConfigTypesFile);
+
+            Assert.assertEquals(expectedUtilsContent, generatedConfigTypesContent);
+
+        } catch (UtilsGenerationException e) {
+            Assert.fail("Error while generating the utils code. " + e.getMessage());
+        }
+    }
+
+    @Test(description = "Test the functionality of the GraphQL utils code generator with API keys config")
+    public void testGenerateSrcWithApiKeysConfig()
+            throws ValidationException, CmdException, IOException, ParseException {
+        try {
+            List<GraphqlProject> projects = TestUtils.getValidatedMockProjects(
+                    this.resourceDir.resolve(Paths.get("specs",
+                            "graphql-config-with-auth-apikeys-config.yaml")).toString(),
+                    this.tmpDir);
+
+            Extension extensions = projects.get(0).getExtensions();
+
+            AuthConfig authConfig = new AuthConfig();
+            AuthConfigGenerator.getInstance().populateAuthConfigTypes(extensions, authConfig);
+            AuthConfigGenerator.getInstance().populateApiHeaders(extensions, authConfig);
+
+            ConfigGenerator.getInstance().generateConfigRecord(authConfig);
+
+            String generatedConfigTypesContent = ConfigGenerator.getInstance().generateSrc(authConfig)
+                    .trim().replaceAll("\\s+", "")
+                    .replaceAll(System.lineSeparator(), "");
+
+            Path expectedConfigTypesFile =
+                    resourceDir.resolve(Paths.get("expectedGenCode", "client", "apiKeysConfig",
+                            "config_types.bal"));
+            String expectedUtilsContent = readContent(expectedConfigTypesFile);
+
+            Assert.assertEquals(expectedUtilsContent, generatedConfigTypesContent);
+
+        } catch (UtilsGenerationException e) {
+            Assert.fail("Error while generating the utils code. " + e.getMessage());
+        }
+    }
+
+    @Test(description = "Test the functionality of the GraphQL utils code generator with API keys config")
```

Comment:
with `client config`

Reply:
Fixed in https://github.com/ballerina-platform/graphql-tools/commit/eb969edf9f95622fe8d6dc7ecc000346998b403e